src/master-playlist-controller.js

```diff
@@ -5,35 +5,133 @@ import PlaylistLoader from './playlist-loader';
 import SegmentLoader from './segment-loader';
 import Ranges from './ranges';
 import videojs from 'video.js';
-import HlsAudioTrack from './hls-audio-track';
 import AdCueTags from './ad-cue-tags';
 
 // 5 minute blacklist
 const BLACKLIST_DURATION = 5 * 60 * 1000;
 let Hls;
 
+/**
+ * determine if an object a is differnt from
+ * and object b. both only having one dimensional
+ * properties
+ *
+ * @param {Object} a object one
+ * @param {Object} b object two
+ * @return {Boolean} if the object has changed or not
+ */
+const objectChanged = function(a, b) {
+  if (typeof a !== typeof b) {
+    return true;
+  }
+  // if we have a different number of elements
+  // something has changed
+  if (Object.keys(a).length !== Object.keys(b).length) {
+    return true;
+  }
+
+  for (let prop in a) {
+    if (!b[prop] || a[prop] !== b[prop]) {
```

Comment:
This will return true if `b[prop]` is a falsely value such as `false` or `0` even if it hasn't changed from `a`